src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/SystemWebAdaptersExtensions.Framework.cs

```diff
@@ -2,61 +2,79 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
-using System.Collections.Generic;
 using System.Web;
+using Microsoft.AspNetCore.SystemWebAdapters;
+using Microsoft.Extensions.DependencyInjection;
 
-namespace Microsoft.AspNetCore.SystemWebAdapters;
+namespace System.Web;
 
 public static class SystemWebAdaptersExtensions
 {
-    private const string Key = "system-web-adapter-builder";
+    private const string BuilderKey = "system-web-adapter-builder";
```

Comment:
Do we want to enforce that the following occurs:

- [x] For any number of calls before building, the same builder is returned
- [ ] Once built, a new builder cannot be created and will fail fast
- [ ] Should someone explicitly call a `Build()` once they are done so it's not "magical"?

I think storing the builder/service provider in the same slot on the `Application` object may make that someone easier.

Reply:
The idea of re-using the same application state slot is interesting. If we make the user call `Build` explicitly, though, then the format of the code for setting up the system web adapter varies more between Net6 and NetFx (the build happens automatically on the Net6 side when the app's service provider is built).

Maybe we should just require a 'build', though, to make it clear.